doc/arch/adr-002-rest-api.md

```diff
@@ -0,0 +1,74 @@
+# Boundary Sync REST API Design
+
+## Context
+
+Based on the [data model](./adr-001-data-models.md), the following REST API design is proposed to access it:
+
+### Authentication
+
+| Path            | Method | Body      | Authentication | Responses                                             | Notes                                        |
+| --------------- | ------ | --------- | -------------- | ----------------------------------------------------- | -------------------------------------------- |
+| `/auth/login/`  | GET    | -         | Allow Any      | 200 OK,<br />401 Not Authorized                       | Returns loging JSON if current session valid |
+| `/auth/login/`  | POST   | `{ ... }` | Allow Any      | 200 OK,<br />400 Bad Request,<br />401 Not Authorized | Used for starting new valid session          |
+| `/auth/logout/` | POST   | -         | Logged In User | 200 OK                                                | Used for logging out                         |
+
+### Boundaries
+
+| Path                                       | Method | Body                     | Authentication | Responses                                                  | Notes                                                                                                              |
+| ------------------------------------------ | ------ | ------------------------ | -------------- | ---------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------ |
+| `/boundaries/`                             | GET    | -                        | Logged In User | 200 OK                                                     | Returns list of all boundaries the user has access to                                                              |
+| `/boundaries/?utilities=1,3,5,8`           | GET    | -                        | Logged In User | 200 OK,<br />401 Not Authorized                            | Returns list of boundaries in the given utilities. If the user does not have access to that utility, returns a 401 |
+| `/boundaries/`                             | POST   | `{ ... }`                | Contributor    | 201 Created,<br />400 Bad Request,<br />401 Not Authorized | Creates a new boundary if the payload is correct and the user is authorized                                        |
+| `/boundaries/{id}/`                        | GET    | -                        | Logged In User | 200 OK,<br />404 Not Found                                 | Returns boundary details if the users has access to it, else 404s. Includes all details about the boundary.        |
```

Comment:
When GETing, this would include:

- The shape
- List of reference layers
- Current status (Draft / Submitted / etc)
- Current Review Annotations
- Activity Log
- Created At, Created By, etc

Reply:
Good catch! Yes I mean reference images.